### ISSUE TITLE
西暦年を意図した`datetime`属性値について、エラーメッセージが適切ではない

### ISSUE THREAD

**Issue by momdo** [markuplint/markuplint#1480]
- Markuplint version: `4.1.0`
- Parser lang: HTML
- Node.js version: `20.11.0`
- OS: Windows

## Describe the bug

整数4桁未満の`datetime`属性値の評価が適切ではありません。
おそらく精度の高い評価は実現困難かと思われます。

仕様：
<https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-time-element:~:text=four%20or%20more>

## Code Example

```html
<time datetime="2">西暦2年</time>
<time datetime="02">西暦2年</time>
<time datetime="002">西暦2年</time>
<time datetime="0002">西暦2年</time><!-- 正しい属性値 -->
```

## Steps To Reproduce

1.  Code Exampleに示すような、`datetime`属性の値が4桁未満の整数を記入します。
 
## The current behavior

1桁、2桁、3桁のエラーメッセージは次のように提示されます。

- 単位の部分がありません。属性「datetime」の単位の部分には「w」「d」「h」「m」「s」のいずれかが必要です (https://html.spec.whatwg.org/multipage/text-level-semantics.html#datetime-value)
- 日時の部分がありません。属性「datetime」の日時の部分にはハイフンが必要です (https://html.spec.whatwg.org/multipage/text-level-semantics.html#datetime-value)
- タイムゾーンの部分がありません。属性「datetime」のタイムゾーンの部分には「Z」「+」「-」のいずれかが必要です (https://html.spec.whatwg.org/multipage/text-level-semantics.html#datetime-value)

## The expected behavior

- （厳密に評価するのであれば、仕様のとおり）「西暦年を提供する場合は、4桁以上の整数（うち1つは0以外の整数）」とする必要があることをメッセージでわかるようにします。もしくは、
- 単に正しくないことだけを伝えるようにします。

**Comment by YusukeHirao** [comment]
@momdo 

なるほど理解しました。たしかに、期待としてはそうなるのはうなずけます。

この問題、警告のルールが[require-datetime](https://markuplint.dev/ja/docs/rules/require-datetime)ではなく、[invalid-attr](https://markuplint.dev/ja/docs/rules/invalid-attr)の評価だからですね。

まず、現状の評価アルゴリズムとしては、属性内のフォーマットに対して行っていて、コンテンツの日付文字列をフォーマットの推論にしていません。
datetime属性の取りうる一番文字数が近いフォーマットのエラーを返します。
たとえば、1桁であれば、一番近い表現は[duration string](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#valid-duration-string)なので、「単位がない」という警告になっています。コンテンツから年表現であることを推論できていないからですね。

どうしようかな…。たしかに実装できなくはないんですが、datetimeだけかなり特別扱いになるんですよね実装上。

**Comment by YusukeHirao** [comment]
#1487 で対応予定
